### PR TITLE
Reroll will accept array of dice objects, remove the check for groupId === string

### PR DESCRIPTION
**Pull request by clevett**
Allow reroll to accept an array of dice objects

Remove the groupId string check from issues #5 

createNotationArray simplified the code. It checks if the parameter is an array and if not makes it one. This makes the logic easier. Also completed the TODO to check for qty & sides. I wrote tests for this before refactoring. All are green.

### CODE REVIEW

File: src/World.js [hunk 1]
```diff
@@ -312,28 +313,31 @@ class World {
 	// accepts array of objects eg: [{sides:int, qty:int, mods:[]}]
 	// accepts object {sides:int, qty:int}
 	createNotationArray(notation){
+		const notation = Array.isArray( input ) ? input : [ input ]
```
Comment (frankieali) [inline src/World.js:316]: `const notation` is already defined by the function argument and `input` is not defined at all. Perhaps you meant
`notation = Array.isArray( notation ) ? notation : [ notation ]`?

File: src/World.js [hunk 2]
```diff
@@ -312,28 +313,31 @@ class World {
...
-				}
-			})
-		} else if(typeof notation === 'object'){
-			// TODO: ensure that there is a 'sides' and 'qty' value on the object - required for making a roll
-			parsedNotation.push(notation)
-		}
+		notation.forEach(roll => {
+			// if notation is an array of strings
+			if ( typeof roll === 'string' ) {
+				parsedNotation.push( parseStringNotation( roll ) )
```
Comment (frankieali) [inline src/World.js:335]: `parseStringNotation` is not defined

**Review by frankieali — CHANGES_REQUESTED** [review]
a few things not working in this PR. Could you take another look?

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
 		"name": "Frank Ali"
 	},
 	"description": "A 3D environment for rolling game dice",
-	"version": "0.4.20",
+	"version": "0.4.30",
```
[the remark this answers is not in the document]
Comment (clevett) [reply package.json]: Changed the version back.

File: src/World.js (hunk 2 above)
Comment (clevett) [reply src/World.js]: I broke out parse in my ts conversion into its own file with tests. This should just be calling parse. Fixed.

File: src/World.js (hunk 1 above)
Comment (clevett) [reply src/World.js:316]: Fix. We're just ensuring the parameter coming in is in an array. Changed the parameter name to input.

File: src/World.js [hunk 4]
```diff
@@ -311,29 +312,32 @@ class World {
...
-				}
-			})
-		} else if(typeof notation === 'object'){
-			// TODO: ensure that there is a 'sides' and 'qty' value on the object - required for making a roll
-			parsedNotation.push(notation)
-		}
+		notation.forEach(roll => {
+			// if notation is an array of strings
+			if ( typeof roll === 'string' ) {
+				parsedNotation.push( parse( roll ) )
```
Comment (frankieali) [inline src/World.js:335]: parse is undefined. I think you mean `this.parse`

**Review by frankieali — CHANGES_REQUESTED** [review]

File: src/World.js (hunk 4 above)
Comment (clevett) [reply src/World.js]: Yep. Fixed! You can tell I am well out of practice with working in classes.

**Pull request merged**